rust-runtime/aws-smithy-http/src/byte_stream/bytestream_util.rs

```diff
@@ -150,32 +171,54 @@ impl FsBuilder {
         let buffer_size = self.buffer_size;
 
         if let Some(path) = self.path {
-            let path_buf = path.to_path_buf();
-            let file_size = self.file_size.unwrap_or(
-                tokio::fs::metadata(path)
+            let offset = self.offset.unwrap_or(DEFAULT_OFFSET);
+            let length = self.length.unwrap_or({
+                let file_length = tokio::fs::metadata(&path)
```

Comment:
seems like this should be `unwrap_or_else`? I'm surprised clippy didn't catch this—this will end up always calling metatdata even when length is specified

Reply:
Yeah, I tried to change it and realized I'd have to do something about the `await` and never got back to it

Reply:
I made a better version based on match